src/gbinder_driver.c

```diff
@@ -835,11 +848,13 @@ gbinder_driver_txstatus(
              */
             switch (tx.status) {
             case (-EAGAIN):
-            case GBINDER_STATUS_FAILED:
-            case GBINDER_STATUS_DEAD_OBJECT:
-                txstatus = (-EFAULT);
                 GWARN("Replacing tx status %d with %d", tx.status, txstatus);
                 break;
+            case GBINDER_STATUS_FAILED:
+                txstatus = GBINDER_STATUS_UNKNOWN_ERROR;
+                break;
+            case GBINDER_STATUS_DEAD_OBJECT:
+                txstatus = (-EPIPE);
```

Comment:
Another break is missing, and isn't this switch supposed to convert things like `INT32_MIN`, `-EPIPE` etc, into the interface values e.g. `GBINDER_STATUS_DEAD_OBJECT`?